extensions/ql-vscode/src/query-history.ts

```diff
@@ -571,12 +599,16 @@ export class QueryHistoryManager extends DisposableObject {
   }
 
   async handleCompareWith(
-    singleItem: FullQueryInfo,
-    multiSelect: FullQueryInfo[]
+    singleItem: LocalQueryInfo,
+    multiSelect: LocalQueryInfo[]
   ) {
     const { finalSingleItem, finalMultiSelect } = this.determineSelection(singleItem, multiSelect);
 
     try {
+      if (finalSingleItem.t !== 'local') {
```

Comment:
Yes, I am planning on using `contextValue`s to determine what is being shown and what isn't. These allow you to control which menu items are shown in which situations. Even with that, it's probably good to keep the checks in place, just in case.